src/e2c/db_connector.py

```diff
@@ -11,31 +12,59 @@
 class DBConnector:
     """Database Connector."""
 
-    def __init__(self, path=_path):
+    def __init__(self, path: str = _path):
         """Initialize database Connector.
 
         Args:
             path (str, optional): Path to the database file. Defaults to _path.
         """
         if pathlib.Path(path).is_file() and path.endswith(".db"):
-            self._connection = DBConnector._init_connection(path)
+            self._conn = DBConnector._init_conn(path)
         else:
             raise ValueError("Database file is missing or invalid.")
 
     @staticmethod
-    def _init_connection(path):
+    def _init_conn(path: str) -> sqlite3.Connection:
         """Initialize database connection."""
         try:
             return sqlite3.connect(path)
         except sqlite3.Error:
             logger.exception("SQLite DB connection failed.")
 
-    def search(self, word):
+    def query(self, word: str = "wrong") -> dict:
         """Search word from the database."""
         try:
             query = "select * from stardict where word = ?"
-            cursor = self._connection.cursor()
+            cursor = self._conn.cursor()
             cursor.execute(query, (word,))
-            return cursor.fetchone()
+            res = cursor.fetchone()
+            res_dict = {
+                "id": res[0],
+                "word": res[1],
+                "sw": res[2],
+                "phonetic": res[3],
+                "definition": res[4],
+                "trans": res[5],
+                "pos": res[6],
+                "collins": res[7],
+                "oxford": res[8],
+                "tag": res[9],
+                "bnc": res[10],
+                "frq": res[11],
+                "exchange": res[12],
```

Comment:
You could define a tuple/array `_key_names` outside `DBConnector` having those names and do
`return [(x, y) for x, y in zip(_key_names, res)]`